### PR TITLE
SWA on 5L/256d + T_max=50 — novel weight-averaging lever

### PR DESCRIPTION
**Pull request by morganmcg1**
## Hypothesis

Stochastic Weight Averaging (SWA; Izmailov et al. 2018, arXiv:1803.05407) collects periodic
weight snapshots in the late phase of training and averages them. The averaged weights tend to
land in a wider, flatter loss basin than any individual checkpoint, leading to better
generalisation without additional training cost. SWA has delivered consistent improvements
across many deep-learning tasks but has **never been tested in this research programme**.

Hypothesis: applying SWA (averaging snapshots from ep40–ep50 every 2 epochs) on the baseline
5L/256d + FourierPE + T_max=50 recipe will reduce val_primary/abupt_axis_mean_rel_l2_pct below
6.9549% by improving generalisation on the wall-shear axes (wsy, wsz).

## Instructions

Implement SWA using PyTorch's built-in `torch.optim.swa_utils`:

```python
from torch.optim.swa_utils import AveragedModel, SWALR

swa_model = AveragedModel(model)
swa_scheduler = SWALR(optimizer, swa_lr=5e-5, anneal_epochs=5)

# From epoch 40 onward, after each epoch:
#   swa_model.update_parameters(model)
#   swa_scheduler.step()
# Update BN stats at end:
#   torch.optim.swa_utils.update_bn(train_loader, swa_model)
```

Run with the standard baseline config (no EMA, no other changes):

```bash
cd target/ && torchrun --standalone --nproc-per-node=4 train.py \
  --model-layers 5 \
  --model-hidden-dim 256 \
  --model-heads 4 \
  --fourier-pe \
  --no-use-ema \
  --lr 3e-4 \
  --lr-cosine-t-max 50 \
  --no-compile-model \
  --agent violet \
  --wandb-group bengio-wave14 \
  --wandb-name violet-swa-lr5e5-ep40-5l256d-tmax50
```

Key implementation notes:
1. Keep the cosine LR schedule running normally from ep1–ep50; SWA scheduler only activates from ep40
2. Collect SWA snapshots every 2 epochs starting at ep40 (i.e. after ep40, ep42, ep44, ep46, ep48, ep50 — 6 total)
3. Log both the **live model** metrics and the **SWA model** metrics from ep40 onward — report both in the PR
4. At ep50, call `torch.optim.swa_utils.update_bn(train_loader, swa_model)` to refresh BN statistics before final eval
5. Report the final SWA model val metrics as your primary result

## Gate schedule (live model — kill if NOT below)

- ep5: val_primary/abupt_axis_mean_rel_l2_pct < 13.0%
- ep10: val_primary/abupt_axis_mean_rel_l2_pct < 10.0%
- ep15: val_primary/abupt_axis_mean_rel_l2_pct < 8.9%
- ep30: val_primary/abupt_axis_mean_rel_l2_pct < 7.5%
- ep50 final SWA model: must beat 6.9549% to WIN

Report wsy and wsz at every gate.

## Baseline (alphonse PR #174, run `vu4jsiic`, ep~45.3)

| Metric | Current Best | AB-UPT Target |
|--------|-------------|---------------|
| `val_primary/abupt_axis_mean_rel_l2_pct` | **6.9549** | 4.51 |
| `val_primary/surface_pressure_rel_l2_pct` | 4.5644 | 3.82 |
| `val_primary/volume_pressure_rel_l2_pct` | 3.9361 ✓ | 6.08 |
| `val_primary/wall_shear_y_rel_l2_pct` | 8.7345 | 3.65 |
| `val_primary/wall_shear_z_rel_l2_pct` | 10.5766 | 3.63 |

## Reference

Izmailov et al. (2018) "Averaging Weights Leads to Wider Optima and Better Generalization"
https://arxiv.org/abs/1803.05407

## ACK requirement

**Reply to this PR within 30 minutes with your W&B run ID to confirm you have started the run.**

### CODE REVIEW

**Comment by morganmcg1** [comment]
ADVISOR: **URGENT — ACK deadline in 7 minutes (by 19:14Z).** No response received yet on PR #446 (SWA on 5L/256d + T_max=50). Start your run immediately and post your W&B run ID. Failure to ACK by 19:14Z will result in this PR being closed and the assignment given to another student.

**Comment by morganmcg1** [comment]
ADVISOR: Closing PR #446 because ADVISOR: PR #446 CLOSED — ACK deadline (19:14Z) missed. No W&B run ID was posted within 30 minutes of assignment, and no response was received to the urgent reminder at 19:07Z. This assignment (SWA on 5L/256d + T_max=50) is being reallocated. violet, you will receive a new assignment in the next wave once you re-engage..
